<launch>
  <!-- <executable cmd="ros2 bag play $(find-pkg-share visp_megapose)/megapose_bag loop"/> -->

  <node pkg="visp_megapose" exec="megapose_client" name="megapose_client" output="screen">
    <!-- set the image topic and camera info topic -->
    <param name="image_topic" value="/camera/camera/color/image_raw"/>
    <param name="camera_info_topic" value="/camera/camera/color/camera_info"/>
    <param name="camera_tf" value="camera_color_optical_frame"/>
    <!-- select the detection method (CLICK or DNN) -->
    <param name="detector_method" value="DNN"/>  
    <!-- set the path to the model directory, format is yolov7 onnx -->
    <param name="detector_model_path" value="$(find-pkg-share visp_megapose)/data/reck.onnx"/>
    <!-- set the DNN detector detect object name -->
    <param name="object_name" value="reck"/>
    <!-- set the Render Object Service Enable (True or False) -->
    <param name="render_enable" value="False"/>
    <!-- set the UI Enable (True or False) -->
    <param name="UI_enable" value="True"/>
  </node>

</launch>
